parlai/core/params.py

```diff
@@ -904,6 +909,13 @@ def add_extra_args(self, args=None):
                 parsed.get('selfchat_task', False),
             )
 
+        # reparse args now that we've inferred some things.  specifically helps
+        # with a misparse of `-opt` as `-o pt` which may be ambigiuous until
+        # after we have added the model args.
```

Comment:
why does adding the model args help with not misparsing?

Reply:
Because before model args are added -opt isn't an argument